core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -147,7 +148,10 @@ private[spark] class CoarseGrainedExecutorBackend(
    * executor exits differently. For e.g. when an executor goes down,
    * back-end may not want to take the parent process down.
    */
-  protected def exitExecutor(code: Int): Unit = System.exit(code)
+  protected def exitExecutor(code: Int, reason: String, throwable: Throwable = null) = {
+    logError(reason, throwable)
```

Comment:
@tejasapatil  do you know if all log frameworks can handle `null` throwable here? Perhaps  it's better to handle it by ourselves like:
```
if (throwable != null) {
   logError(reason, throwable)
} else {
  logError(reason)
}
```